DEVELOPER_GUIDE.md

```diff
@@ -1,29 +1,44 @@
 # DEVELOPER GUIDE
 
+### Extensions
 
+This library defines many extensions for different types.
+Some of these functions are conceptually connected, in that they represent an abstraction and sometimes they have indeed a generic version of the function which doesn't require typing the type as module prefix.
+Note that in these cases, normally the names are the same but this is not a rule, there are cases where the name differs to avoid collisions or confusion with other functions, for example:
```

Comment:
You gave a very interesting explanation of relation of F#+ and F# core library on slack. I can't search it, because I don't have premium Slack version, but it went something along the lines of:
"In F#+ we try to expand FSharp.Core functionalities, but we are trying not to be intrusive"
There is a remark on this in general readme
>F#+ builds upon FSharp, using generic programming techniques to help you avoid boiler plate code. However, by naming conventions and signatures it can be seen to 'enhance' rather than 'replace' existing patterns as much as possible.

But I wonder if we could refer to this concept in a more concrete and contributor-oriented way here.

Reply:
Yes, it's no longer available. But I think what I said was basically:

Reply:
I think, that information from above post should be put inside Developers Guide. 
Currently, dev guide consist mostly of implementation explanation. I think it is a good idea to show high-level "philosophy of F#+" and it's relation to core F# library.  It would make it easier for a newcomer to be aligned with the purpose of this project. It would also give a direction on how we should expand it.